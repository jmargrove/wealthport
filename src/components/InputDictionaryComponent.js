import React, { Component } from "react";
import CloseIcon from "mdi-react/CloseIcon";
import { connect } from "react-redux";
import CheckCircleOutlineIcon from "mdi-react/CheckCircleOutlineIcon";
import { addNewDictionary } from "./../actions/actions.js";
import {
  DelBox,
  ClickBox,
  InputDictionaryContainer
} from "./../presentational/Containers.js";

const mapDispatchToProps = dispatch => ({
  addNewDictionary: dictionaryName => dispatch(addNewDictionary(dictionaryName))
});

class InputDictionaryComponent extends Component {
  constructor (props) {
    super(props);
    this.state = {
      toggleAddDictionaryInput: false
    };
  }

  handleAddNewDictionary = () => {
    this.props.addNewDictionary(this.input.value);
    this.props.toggleAddDictionary();
  };

  handleKeyPress = e => {
    if (e.key === "Enter") {
      this.props.addNewDictionary(this.input.value);
      this.props.toggleAddDictionary();
    }
  };

  render () {
    return (
      <InputDictionaryContainer className="InputDictionaryContainer">
        <input
          style={{ backgroundColor: "#E0F1FF" }}
          onKeyPress={this.handleKeyPress}
          autoFocus={true}
          onBlur={this.props.toggleAddDictionary}
          ref={comp => (this.input = comp)}
          placeholder="Add..."
        />
        <DelBox onMouseDown={this.props.toggleAddDictionary}>
          <CloseIcon />
        </DelBox>
        <ClickBox onMouseDown={this.handleAddNewDictionary}>
          <CheckCircleOutlineIcon />
        </ClickBox>
      </InputDictionaryContainer>
    );
  }
}

export default connect(null, mapDispatchToProps)(InputDictionaryComponent);
