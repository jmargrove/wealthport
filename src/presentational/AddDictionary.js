import React, { Component } from "react";
import { AddDictionaryContainer } from "./Containers.js";
import InputDictionaryComponent from "./../components/InputDictionaryComponent.js";

class AddDictionary extends Component {
  constructor (props) {
    super(props);
    this.state = {
      toggleAddDictionaryInput: false
    };
  }

  toggleAddDictionary = () => {
    this.setState((prevState, props) => ({
      toggleAddDictionaryInput: !prevState.toggleAddDictionaryInput
    }));
  };

  render () {
    if (this.state.toggleAddDictionaryInput) {
      return (
        <InputDictionaryComponent
          toggleAddDictionary={this.toggleAddDictionary}
        />
      );
    } else {
      return (
        <AddDictionaryContainer onClick={this.toggleAddDictionary}>
          <p> Add Dictionary </p>
        </AddDictionaryContainer>
      );
    }
  }
}

export default AddDictionary;
