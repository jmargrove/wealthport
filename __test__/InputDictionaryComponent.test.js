import React from "react";
import { shallow, mount, configure } from "enzyme";
import renderer from "react-test-renderer";
import InputDictionaryComponent from "./../src/components/InputDictionaryComponent.js";
import AddDictionary from "./../src/presentational/AddDictionary.js";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { createStore } from "redux";
import calculatorReducers from "./../src/reducer.js";
import Adapter from "enzyme-adapter-react-15";
import ShallowRenderer from "react-test-renderer/shallow";

configure({ adapter: new Adapter() });

describe(">>>InputDictionaryComponent  +++capturing Snapshot of InputDictionaryComponent", () => {
  const mockStore = configureStore();
  let store;

  it("+++capturing Snapshot of InputDictionaryComponent", () => {
    store = createStore(calculatorReducers);
    const renderedValue = renderer
      .create(
        <Provider store={store}>
          <InputDictionaryComponent />
        </Provider>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

//
describe(">>>InputDictionaryComponent +++ testing function --- toggleAddDictionary", () => {
  let store;
  store = createStore(calculatorReducers);
  it("+++ testing toggleAddDictionary return undefined", () => {
    store = createStore(calculatorReducers);
    const renderedValue = renderer
      .create(
        <Provider store={store}>
          <InputDictionaryComponent />
        </Provider>
      )
      .toJSON();

    const handleKeyPress = renderedValue.children[0].props.onKeyPress;
    expect(handleKeyPress("Enter")).toBe(undefined);
    expect(handleKeyPress("blah")).toBe(undefined);
  });

  it("+++ testing autoFocus on the input box is always TRUE", () => {
    store = createStore(calculatorReducers);
    const renderedValue = renderer.create(
      <Provider store={store}>
        <InputDictionaryComponent />
      </Provider>
    ).root;
    const theState = renderedValue._fiber.stateNode.props.store.getState();
    expect(renderedValue.findAllByType("div").length).toBe(3);
    expect(renderedValue.findAllByType("input").length).toBe(1);
  });

  it("+++ testing if the viewDictionary is a string", () => {
    store = createStore(calculatorReducers);
    const renderedValue = renderer.create(
      <Provider store={store}>
        <InputDictionaryComponent />
      </Provider>
    ).root;
    const theState = renderedValue._fiber.stateNode.props.store.getState();
    expect(typeof theState.viewDictionary).toBe("string");
  });

  it("+++ testing if dictionaries are stored in a object", () => {
    store = createStore(calculatorReducers);
    const renderedValue = renderer.create(
      <Provider store={store}>
        <InputDictionaryComponent />
      </Provider>
    ).root;
    const theState = renderedValue._fiber.stateNode.props.store.getState();
    expect(typeof theState.dictionaries).toBe("object");
  });

  it("+++ testing if the test type is a sting", () => {
    store = createStore(calculatorReducers);
    const renderedValue = renderer.create(
      <Provider store={store}>
        <InputDictionaryComponent />
      </Provider>
    ).root;
    const theState = renderedValue._fiber.stateNode.props.store.getState();
    expect(typeof theState.testType).toBe("string");
  });
});
