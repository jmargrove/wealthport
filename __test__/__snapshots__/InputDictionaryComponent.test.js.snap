// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>InputDictionaryComponent  +++capturing Snapshot of InputDictionaryComponent +++capturing Snapshot of InputDictionaryComponent 1`] = `
<div
  className="InputDictionaryContainer sc-bxivhb fsdWwv"
>
  <input
    autoFocus={true}
    onBlur={undefined}
    onKeyPress={[Function]}
    placeholder="Add..."
    style={
      Object {
        "backgroundColor": "#E0F1FF",
      }
    }
  />
  <div
    className="sc-bwzfXH iTItjm"
    onMouseDown={undefined}
  >
    <svg
      className="mdi-icon "
      fill="#000"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
      />
    </svg>
  </div>
  <div
    className="sc-bdVaJa jovjJ"
    onMouseDown={[Function]}
  >
    <svg
      className="mdi-icon "
      fill="#000"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12,2C17.52,2 22,6.48 22,12C22,17.52 17.52,22 12,22C6.48,22 2,17.52 2,12C2,6.48 6.48,2 12,2M12,4C7.58,4 4,7.58 4,12C4,16.42 7.58,20 12,20C16.42,20 20,16.42 20,12C20,7.58 16.42,4 12,4M11,16.5L6.5,12L7.91,10.59L11,13.67L16.59,8.09L18,9.5L11,16.5Z"
      />
    </svg>
  </div>
</div>
`;
